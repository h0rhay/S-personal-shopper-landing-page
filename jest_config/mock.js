export const mockTheme = {
  themeType: 'dark',
  breakpoints: {
    small: '(max-width: 812px)',
    medium: '(min-width: 813px)',
    large: '(min-width: 960px)',
    deviceSwitchSm: '(max-width: 896px)',
    deviceSwitchLg: '(min-width: 897px)',
  },
  typography: {
    mobile: {
      headerOne: {
        fontSize: '3rem',
      },
      headerTwo: {
        fontSize: '1.75rem',
      },
      displayHeading: {
        fontSize: 'calc(1.5rem * 2)',
      },
      itemHeading: {
        fontSize: '1.125rem',
      },
    },
    desktop: {
      headerOne: {
        fontSize: '4rem',
      },
      headerTwo: {
        fontSize: '2rem',
      },
      displayHeading: {
        fontSize: 'calc(4rem * 2)',
      },
      itemHeading: {
        fontSize: '1.125rem',
      },
    },
  },
  text: '#ffffff',
  primary: '#212121',
  primaryAccent: '#2E2E2E',
  accent: '#ffe256',
  shadow: 'rgba(0, 0, 0, 0.4)',
};
