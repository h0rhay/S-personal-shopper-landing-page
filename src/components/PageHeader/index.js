import PropTypes from 'prop-types';
import React from 'react';
import styled, { css } from 'styled-components';

import SelfridgesLogo from '../../images/selfridges_logo.svg';
import SiteWidthWrapper from '../SiteWidthWrapper';

const Header = styled.header`
  position: fixed;
  top: 0;
  z-index: 10;
  ${({ theme }) =>
    theme.themeType === 'dark'
      ? css`
          background: ${theme.primaryAccent};
        `
      : css`
          background: ${theme.accent};
        `}

  height: 3.875rem;
  width: 100%;
  display: flex;
  justify-content: space-around;
  align-items: center;
  margin-top: 0;
  svg {
    display: block;
    max-width: 9.065rem;
  }
`;

const SiteLogo = styled(SelfridgesLogo)`
  path {
    ${({ theme }) =>
      theme.themeType === 'dark'
        ? css`
            fill: ${theme.text};
          `
        : css`
            fill: ${theme.primary};
          `}
  }
`;

const Nav = styled.nav`
  margin-top: 0;
  display: flex;
  justify-content: space-between;
  * {
    ${({ theme }) =>
      theme.themeType === 'dark'
        ? css`
            color: ${theme.text};
          `
        : css`
            color: ${theme.primary};
          `}
  }
`;

const HeaderLinkWrap = styled.div`
  display: flex;
  align-items: center;
`;

const Contact = styled.a`
  text-transform: uppercase;
  margin-left: 2rem;
  ${({ theme }) =>
    theme.themeType === 'dark'
      ? css`
          color: ${theme.text};
        `
      : css`
          color: ${theme.primary};
        `}
  &:focus, &:visited {
    ${({ theme }) =>
      theme.themeType === 'dark'
        ? css`
            color: ${theme.text};
          `
        : css`
            color: ${theme.primary};
          `}
  }
`;

const NavLink = styled.a`
  span {
    display: none;
  }
`;

const PageHeader = ({ contact }) => (
  <Header>
    <SiteWidthWrapper>
      <Nav>
        <NavLink
          data-testid="selfridges-logo-link"
          href="https://www.selfridges.com/"
          target="_blank"
          rel="noopener noreferrer"
          title="Selfridges &amp; Co"
        >
          <SiteLogo />
          <span>Selfridges &amp; Co</span>
        </NavLink>
        <HeaderLinkWrap>
          <Contact
            data-testid="contact-me-link"
            href={`mailto:${contact}?subject=Personal Shopping Appointment`}
            target="_blank"
          >
            Contact Me
          </Contact>
        </HeaderLinkWrap>
      </Nav>
    </SiteWidthWrapper>
  </Header>
);

PageHeader.propTypes = {
  contact: PropTypes.string,
};

export default PageHeader;
