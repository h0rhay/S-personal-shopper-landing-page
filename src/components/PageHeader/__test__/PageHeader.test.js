import React from 'react';
import { render, screen } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { mockTheme } from '../../../../jest_config/mock';
import PageHeader from '../index';

let container;
const props = {
  contact: 'test@test.co.uk',
};

const renderComponent = ({ theme }) =>
  render(
    <ThemeProvider theme={theme}>
      <PageHeader {...props} />
    </ThemeProvider>
  );

describe('Page Header', () => {
  beforeEach(() => {
    container = renderComponent({ theme: mockTheme }).container;
  });
  it('should match screenshot', () => {
    expect(container).toMatchSnapshot();
  });
  it('should contain email in contact me href', () => {
    const contact = screen.getByTestId('contact-me-link');
    expect(contact.href).toContain(props.contact);
  });
  it('should contain selfridges url in logo', () => {
    const logo = screen.getByTestId('selfridges-logo-link');
    expect(logo.href).toContain('https://www.selfridges.com/');
  });
});
