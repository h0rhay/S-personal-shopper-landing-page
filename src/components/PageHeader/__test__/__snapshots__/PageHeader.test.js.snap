// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Header should match screenshot 1`] = `
<div>
  <header
    class="sc-gsDKAQ eWeztX"
  >
    <div
      class="sc-bdvvtL kiJDep site_width_wrap"
    >
      <nav
        class="sc-hKwDye ihRzwE"
      >
        <a
          class="sc-gKclnd kpKMdZ"
          data-testid="selfridges-logo-link"
          href="https://www.selfridges.com/"
          rel="noopener noreferrer"
          target="_blank"
          title="Selfridges & Co"
        >
          <test-file-stub
            classname="sc-dkPtRN fFrQVd"
          />
          <span>
            Selfridges & Co
          </span>
        </a>
        <div
          class="sc-eCImPb cgYwop"
        >
          <a
            class="sc-jRQBWg gqQVrW"
            data-testid="contact-me-link"
            href="mailto:test@test.co.uk?subject=Personal Shopping Appointment"
            target="_blank"
          >
            Contact Me
          </a>
        </div>
      </nav>
    </div>
  </header>
</div>
`;
